threestudio/systems/gaussian_splatting.py

```diff
@@ -0,0 +1,256 @@
+    return np.float32(Rt)
+
+def getProjectionMatrix(znear, zfar, fovX, fovY):
+    tanHalfFovY = math.tan((fovY / 2))
+    tanHalfFovX = math.tan((fovX / 2))
+
+    top = tanHalfFovY * znear
+    bottom = -top
+    right = tanHalfFovX * znear
+    left = -right
+
+    P = torch.zeros(4, 4)
+
+    z_sign = 1.0
+
+    P[0, 0] = 2.0 * znear / (right - left)
+    P[1, 1] = 2.0 * znear / (top - bottom)
+    P[0, 2] = (right + left) / (right - left)
+    P[1, 2] = (top + bottom) / (top - bottom)
+    P[3, 2] = z_sign
+    P[2, 2] = z_sign * zfar / (zfar - znear)
+    P[2, 3] = -(zfar * znear) / (zfar - znear)
+    return P
+
+def get_cam_info(c2w, fovy):
+    matrix = np.linalg.inv(c2w[0].cpu().numpy())
+    R = np.transpose(matrix[:3,:3])
+    R[:,0] = -R[:,0]
+    T = -matrix[:3, 3]
+    
+    world_view_transform = torch.tensor(getWorld2View2(R, T)).transpose(0, 1).cuda()
+    projection_matrix = getProjectionMatrix(znear=0.01, zfar=100, fovX=fovy, fovY=fovy).transpose(0,1).cuda()
+    full_proj_transform = (world_view_transform.unsqueeze(0).bmm(projection_matrix.unsqueeze(0))).squeeze(0)
+    camera_center = world_view_transform.inverse()[3, :3]
+    
+    return world_view_transform, full_proj_transform, camera_center
+
+class Camera(NamedTuple):
+    FoVx: torch.Tensor
+    FoVy: torch.Tensor
+    camera_center: torch.Tensor
+    image_width: int
+    image_height: int
+    world_view_transform: torch.Tensor
+    full_proj_transform: torch.Tensor
+        
+@threestudio.register("gaussian-splatting-system")
+class GaussianSplatting(BaseLift3DSystem):
+    @dataclass
+    class Config(BaseLift3DSystem.Config):
+        extent: float = 5.0
+        num_pts: int = 100
+        invert_bg_prob: float = 0.5
+
+    cfg: Config
+
+    def configure(self) -> None:
+        # set up geometry, material, background, renderer
+        super().configure()
+        self.automatic_optimization=False
+        
+        self.background_tensor = torch.tensor(
+            [1, 1, 1], 
+            dtype=torch.float32, 
+            device="cuda"
+        )
+        # Since this data set has no colmap data, we start with random points
+        num_pts = self.cfg.num_pts
+        print(f"Generating random point cloud ({num_pts})...")
+        self.extent = self.cfg.extent
+        phis = np.random.random((num_pts,)) * 2 * np.pi
+        costheta = np.random.random((num_pts,)) * 2 - 1
+        thetas = np.arccos(costheta)
+        mu = np.random.random((num_pts,))
+        radius = 0.5 * np.cbrt(mu)
+        x = radius * np.sin(thetas) * np.cos(phis)
+        y = radius * np.sin(thetas) * np.sin(phis)
+        z = radius * np.cos(thetas)
+        xyz = np.stack((x, y, z), axis=1)
+
+        shs = np.random.random((num_pts, 3)) / 255.0
+        C0 = 0.28209479177387814
+        color = shs * C0 + 0.5
+        pcd = BasicPointCloud(
+            points=xyz, colors=color, normals=np.zeros((num_pts, 3))
+        )
+        
+        self.geometry.create_from_pcd(pcd, 10)
```

Comment:
Considering the generation process as an initialization step, similar to how we handle implicit volume and SDF initialization. Maybe we could relocate it into gaussian.py for a cleaner structure? What do you think?